File: LeetCode/Easy/0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int minDiff = Integer.MAX_VALUE;
    Integer prev = null;
    public int getMinimumDifference(TreeNode root) {
        inorder(root);
        return minDiff;
    }
    public void inorder(TreeNode node) {
        if (node == null) {
            return;
        }
        inorder(node.left);
        if (prev == null) {
            prev = node.val;
        } else {
            minDiff = Math.min(minDiff, node.val - prev);
            prev = node.val;
        }
        inorder(node.right);
    }
}

File: LeetCode/Easy/0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func getMinimumDifference(root *TreeNode) int {
    ans := 100000
    nums := make([]int, 0)

    q := make([]*TreeNode, 0)
    q = append(q, root)
    nums = append(nums, root.Val)

    for (len(q) != 0) {
        cur := q[0]
        q = q[1:]

        if (cur.Left != nil) {
            q = append(q, cur.Left)
            nums = append(nums, cur.Left.Val)
        }

        if (cur.Right != nil) {
            q = append(q, cur.Right)
            nums = append(nums, cur.Right.Val)
        }
    }
    sort.Ints(nums)
    for i := 0; i < len(nums) - 1; i++ {
        ans = min(ans, nums[i+1] - nums[i])
    }
    return ans
}